token-lending/program/src/processor.rs

```diff
@@ -578,6 +587,309 @@ fn _deposit_reserve_liquidity<'a>(
     Ok(collateral_amount)
 }
 
+fn process_flash_borrow_reserve_liquidity(
+    program_id: &Pubkey,
+    liquidity_amount: u64,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+    if liquidity_amount == 0 {
+        msg!("Liquidity amount provided cannot be zero");
+        return Err(LendingError::InvalidAmount.into());
+    }
+
+    let account_info_iter = &mut accounts.iter();
+    let source_liquidity_info = next_account_info(account_info_iter)?;
+    let destination_liquidity_info = next_account_info(account_info_iter)?;
+    let reserve_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let lending_market_authority_info = next_account_info(account_info_iter)?;
+    let sysvar_info = next_account_info(account_info_iter)?;
+    let token_program_id = next_account_info(account_info_iter)?;
+
+    // We don't care about the return value here, so just ignore it.
+    _flash_borrow_reserve_liquidity(
+        program_id,
+        liquidity_amount,
+        source_liquidity_info,
+        destination_liquidity_info,
+        reserve_info,
+        lending_market_info,
+        lending_market_authority_info,
+        sysvar_info,
+        token_program_id,
+    )?;
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+fn _flash_borrow_reserve_liquidity<'a>(
```

Comment:
- [ ] We probably want a way to disable flash borrowing for a market (or the protocol) altogether.

Reply:
maybe we could just have it be that if fee is set to 100% it's disabled?